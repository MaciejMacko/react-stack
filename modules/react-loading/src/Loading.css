.react-loading {
  display: inline-block;
  text-align: center;
}

.react-loading .icon {
  font-size: 4em;
}

.react-loading .react-loading--text {
  margin-top: 1em;
  text-transform: uppercase;
}


.react-loading--inline .icon {
  font-size: 1em;
}

.react-loading--inline .react-loading--text {
  display: inline-block;
  margin-top: 0;
  margin-left: 0.5em;
}
