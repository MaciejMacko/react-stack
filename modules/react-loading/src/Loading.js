import './Loading.css'
import React from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'
import Icon from '@terraeclipse/react-icon'

class Loading extends React.Component {

  static propTypes = {
    item: PropTypes.string,
    text: PropTypes.string,
    inline: PropTypes.bool
  }

  static defaultProps = {
    item: '',
    text: '',
    inline: false
  }

  render () {
    let {item, text, inline} = this.props
    return (
      <div className={cn('react-loading', {'react-loading--inline': inline})}>
        <Icon type='circle-o-notch' className='fa-spin' />
        {(item.length || text.length) ? (
          <div className='react-loading--text'>
            {`${text.length ? text : 'Loading'}${item.length ? ' ' : ''}${item}...`}
          </div>
        ) : null}
      </div>
    )
  }
}

export default Loading
