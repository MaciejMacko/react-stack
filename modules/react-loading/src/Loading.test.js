/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import Loading from './'

describe('Loading', () => {
  it('renders', () => {
    const component = renderer.create(<Loading />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
