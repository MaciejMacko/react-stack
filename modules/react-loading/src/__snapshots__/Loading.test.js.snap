// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders 1`] = `
<div
  className="react-loading"
>
  <i
    className="icon fa fa-circle-o-notch fa-spin"
    onClick={undefined}
  />
</div>
`;
